tuplex/core/src/physical/ResolveTask.cc

```diff
@@ -817,113 +867,186 @@ namespace tuplex {
+                    numInputRowsLeftInPartition = _inputExceptions[curInputPartitionInd]->getNumRows();
+                    inputPtr = _inputExceptions[curInputPartitionInd]->lock();
+                } else {
+                    // processed all exceptions
+                    inputPtr = nullptr;
+                }
+            }
+        }
 
-            // merge exceptions with normal rows after calling slow code over them...
-            // basic idea is go over all exception partitions, execute row wise the resolution function
-            // and merge the result back to the partitions
-            for(auto partition : _exceptions) {
-                const uint8_t* ptr = partition->lockRaw();
-                int64_t numRows = *((int64_t*)ptr); ptr += sizeof(int64_t);
+        // add remaining normal rows & partitions to merged partitions
+        while(_normalRowNumber < _normalNumRows) {
+            // trick: to get row size, you know number of normal elements + variable length!
+            // ==> can be used for quick merging!
+            size_t size = readOutputRowSize(_normalPtr, _normalPtrBytesRemaining);
 
-                for(int i = 0; i < numRows; ++i) {
+            writeRow(_normalPtr, size);
+            _normalPtr += size;
+            _normalPtrBytesRemaining -= size;
+            _normalRowNumber++;
+        }
 
-                    // old
-                    // _currentRowNumber = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    // int64_t ecCode = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    // int64_t operatorID = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    // int64_t eSize = *((int64_t*)ptr);
-                    // ptr += sizeof(int64_t);
-                    const uint8_t *ebuf = nullptr;
-                    int64_t ecCode = -1, operatorID = -1;
-                    size_t eSize = 0;
-                    auto delta = deserializeExceptionFromMemory(ptr, &ecCode, &operatorID, &_currentRowNumber, &ebuf,
-                                                                &eSize);
-                    processExceptionRow(ecCode, operatorID, ebuf, eSize);
-                    ptr += delta;
-                    // old
-                    // ptr += eSize;
-                }
-                partition->unlock();
-            }
+        if (!_partitions.empty())
+            _partitions[_currentNormalPartitionIdx]->unlock();
 
+        // merging is done, unlock the last partition & copy the others over.
+        unlockAll();
 
-            // done, unlock all partitions
-            unlockAll();
+        for(int i = _currentNormalPartitionIdx + 1; i < _partitions.size(); ++i) {
+            _mergedRowsSink.unlock();
+            _mergedRowsSink.partitions.push_back(_partitions[i]);
+            //_mergedPartitions.emplace_back(_normalCasePartitions[i]);
```

Comment:
what's up with this comment? can we remove the line or should be commented back in?